### PR TITLE
adding perplexity to bert validation

### PR DESCRIPTION
**Pull request by yzhang123**
Signed-off-by: Yang Zhang <yangzhang@nvidia.com>

### CODE REVIEW

File: nemo/collections/nlp/metrics/__init__.py [hunk 1]
```diff
@@ -0,0 +1,2 @@
+from nemo.collections.nlp.metrics.classification_report import ClassificationReport
```
Comment (VahidooX) [inline nemo/collections/nlp/metrics/__init__.py:1]: header file missing.

File: examples/nlp/bert_pretraining_from_text.py [hunk 2]
```diff
@@ -42,17 +42,18 @@ def add_args(parser):
     )
     parser.add_argument(
         "--gpus",
+        type=int,
```
Comment (VahidooX) [inline examples/nlp/bert_pretraining_from_text.py:45]: Why is it still based on args?

File: nemo/collections/nlp/metrics/perplexity.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+class Perplexity(TensorMetric):
+    """
+    This metric computes the perplexity given the language model loss.
+    """
+
+    def __init__(self):
+        super(Perplexity, self).__init__(name="Perplexity")
+
+    def forward(self, loss: torch.Tensor) -> torch.Tensor:
+        return torch.exp(loss).to(loss.device)
```
Comment (VahidooX) [inline nemo/collections/nlp/metrics/perplexity.py:34]: Do we need .to here as torch.exp would keep the device?

File: nemo/collections/nlp/models/lm_model.py [hunk 4]
```diff
@@ -154,7 +158,8 @@ def validation_step(self, batch, batch_idx):
 
         loss = self.agg_loss(loss_1=mlm_loss, loss_2=nsp_loss)
 
-        tensorboard_logs = {'val_loss': loss}
+        perplexity = self.perplexity(mlm_loss)
```
Comment (VahidooX) [inline nemo/collections/nlp/models/lm_model.py:161]: how about changing the name of var perplexity to perp_val or the name of perplexity to perplexity_metric?

File: nemo/collections/nlp/metrics/__init__.py (hunk 1 above)
Comment (yzhang123) [reply nemo/collections/nlp/metrics/__init__.py:1]: done

File: examples/nlp/bert_pretraining_from_text.py (hunk 2 above)
Comment (yzhang123) [reply examples/nlp/bert_pretraining_from_text.py:45]: Eric is working on Hydra

File: nemo/collections/nlp/metrics/perplexity.py (hunk 3 above)
Comment (yzhang123) [reply nemo/collections/nlp/metrics/perplexity.py:34]: done

File: nemo/collections/nlp/models/lm_model.py (hunk 4 above)
Comment (yzhang123) [reply nemo/collections/nlp/models/lm_model.py:161]: i like perplexity_metric

**Pull request merged**